specification/under_development/profiles/Base_Profile.md

```diff
@@ -0,0 +1,484 @@
+
+The entry point may not take any parameters and must return an exit code in the
+form of a 64-bit integer. The exit code `0` must be used to indicate a
+successful execution of the program.
+
+### Attributes
+
+The following custom attributes must be attached to the entry point function:
+
+- An attribute named `"entry_point"` identifying the function as the starting
+  point of a quantum program
+- An attribute name `"qir_profile"` with the value `"base_profile"` identifying
+  the profile the entry point has been compiled for
+- An attribute name `"output_labels"` with an arbitrary string value that
+  identifies the schema used by a [compiler
+  frontend](https://en.wikipedia.org/wiki/Compiler#Front_end) that produced the
+  IR to label the recorded output
+- An attribute named `"required_qubits"` indicating the number of qubits used by
+  the entry point
+- An attribute named `"required_results"` indicating the maximal number of
+  measurement results that need to be stored while executing the entry point
+  function
+
+Optionally, additional attributes may be attached to the entry point. Any custom
+function attributes attached to the entry point will be reflected as metadata in
+the program output; this includes both mandatory and optional attributes but not
+parameter attributes or return value attributes. This in particular implies that
+the [labeling schema](#output-recording) used in the recorded output can be
+identified by looking at the metadata in the produced output. See the
+specification of the [output schemas](../output_schemas/) for more information
+about how metadata is represented in the output schema.
+
+Custom function attributes will show up as part of an [attribute
+group](https://releases.llvm.org/13.0.1/docs/LangRef.html#attrgrp) in the IR.
+Attribute groups are numbered such that they can be easily referenced by
+multiple function definitions or global variables. Consumers of Base Profile
+compliant programs must not rely on a particular numbering, but instead look for
+function to which an attribute with the name `"entry_point"` is attached to
+determine which one to invoke when the program is launched.
+
+Within the restrictions imposed by the Base Profile, the number of qubits that
+are needed to execute the quantum program must be known at compile time. This
+number is captured in the form of the `"required_qubits"` attribute attached to
+the entry point. The value of the attribute must be the string representation of
+a 64-bit integer constant.
+
+Similarly, the number of measurement results that need to be stored when
+executing the entry point function is captured by the `"required_results"`
+attribute. Since qubits cannot be used after measurement, in the case of the
+Base Profile, this value is equal to the number of measurement results in the
+program output.
+
+For a program to be valid, both the number of qubits used and the number of
```

Comment:
=> both `"required_qubits"` and `"required_results"` must be strictly positive? (makes it more straightforward)

Reply:
Same as my comment below; I find it confusing to use the name of the attribute when talking about the value of the attribute.